packages/select/src/index.mts

```diff
@@ -81,14 +93,18 @@ export default createPrompt<string, SelectConfig>((config, done) => {
   }
 
   const allChoices = choices
-    .map(({ name, value, disabled }, index) => {
+    .map(({ name, value, disabled, type, separator }, index) => {
       const line = name || value;
       if (disabled) {
         return chalk.dim(
           `- ${line} ${typeof disabled === 'string' ? disabled : '(disabled)'}`
         );
       }
 
+      if (type === 'separator') {
+        return ` ${separator}`;
```

Comment:
Do we need/should we force a whitespace prefix? (genuinely asking if that's what most users would expect; I'm not sure)